### PR TITLE
First pass implementation of bit:partition and fn:ends-with

### PR DESCRIPTION
**Pull request by AlvinKuruvilla**
This commit does a first pass implementation of bit:partition, which does not implement the functionality but sketches it out. Fn:ends-with also is implemented with its functionality

### CODE REVIEW

File: src/main/java/org/brackit/xquery/function/fn/EndsWith.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    public Sequence execute(StaticContext sctx, QueryContext ctx, Sequence[] args) {
+        if (isZeroLength(args) == Boolean.TRUE) {
+            return Int32.ONE;
+        }
+        if (isZeroLength(args) == Boolean.FALSE) {
+            return Int32.ZERO;
+        }
+        String first = ((Item) args[0]).atomize().stringValue();
+        String second = ((Item) args[1]).atomize().stringValue();
+        boolean result = first.contains(second);
```
Comment (JohannesLichtenberger) [inline src/main/java/org/brackit/xquery/function/fn/EndsWith.java:40]: Seems to be strange, that you compare the strings with contains, but the function is called `fn:ends-with`. Second, it should return a Boolean value. Third, the function has to be registered and test cases would be nice :-)

Comment (AlvinKuruvilla) [reply src/main/java/org/brackit/xquery/function/fn/EndsWith.java]: Yeah looking at it again using `contains()` was silly but that should be fixed in my newest commit. In regards to your second point about Boolean value... my initial implementation returned `Boolean.TRUE` and `Boolean.FALSE` but there is a type mismatch with the expected return type Sequence so I switched to Int32

Comment (AlvinKuruvilla) [reply src/main/java/org/brackit/xquery/function/fn/EndsWith.java]: Regarding your last point: I will add a test case as soon as I can

Comment (JohannesLichtenberger) [reply src/main/java/org/brackit/xquery/function/fn/EndsWith.java]: You need the type `Bool` which implements `Sequence`. Here are examples you can use for your tests:

https://www.w3.org/TR/xpath-functions-31/#func-ends-with

**Pull request closed without merging**